bv-evaluation/compare.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import os
+import subprocess
+import concurrent.futures
+import shutil
+
+bv_widths = [4, 8, 16, 32, 64]
+
+ROOT_DIR = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode('utf-8').strip()
+
+RESULTS_DIR_ALIVE = ROOT_DIR + '/bv-evaluation/results/Alive/'
+BENCHMARK_DIR_ALIVE = ROOT_DIR + '/SSA/Projects/InstCombine/'
+
+RESULTS_DIR_HACKERSDELIGHT = ROOT_DIR + '/bv-evaluation/results/HackersDelight/'
+BENCHMARK_DIR_HACKERSDELIGHT = ROOT_DIR + '/SSA/Projects/InstCombine/HackersDelight/'
+
+RESULTS_DIR_INSTCOMBINE = ROOT_DIR + '/bv-evaluation/results/InstCombine/'
+BENCHMARK_DIR_INSTCOMBINE = ROOT_DIR + '/SSA/Projects/InstCombine/tests/proofs/'
+
+TIMEOUT = 1800
+REPS = 1
+
+def clear_folder(results_dir):
+    """Clears all files and subdirectories within the given directory."""
+    if not os.path.exists(results_dir):
+        return
+
+    for item in os.listdir(results_dir):
+        item_path = os.path.join(results_dir, item)
+        try:
+            if os.path.isfile(item_path) or os.path.islink(item_path):
+                os.unlink(item_path)
+            elif os.path.isdir(item_path):
+                shutil.rmtree(item_path)
+        except Exception as e:
+            print(f'Failed to delete {item_path}. Reason: {e}')
+
+def run_file(benchmark : str, file_to_run : str, log_file_base_name : str, specific_arg_for_log_name: str = None) :
+    """
+    Runs a single Lean file and logs its output.
+    file_to_run: The full path to the .lean file to execute.
+    log_file_base_name: The base name for the log file.
+    specific_arg_for_log_name: An optional argument to append to the log file name, useful for hackersdelight bit-width.
+    """
+    cmd_prefix = 'lake lean '
+    if benchmark == "hackersdelight":
+        try:
+            for r in range(REPS):
+                # Construct log file name including original file's base name and bit-width
+                log_file_name = f'{log_file_base_name}_{specific_arg_for_log_name}_r{r}.txt'
+                log_file_path = os.path.join(RESULTS_DIR_HACKERSDELIGHT, log_file_name)
+                with open(log_file_path, 'w') as log_file:
+                    cmd = cmd_prefix + file_to_run
+                    print(f"Running: {cmd}")
+                    try:
+                        subprocess.Popen(cmd, cwd=ROOT_DIR, stdout=log_file, stderr=log_file, shell=True).wait(timeout=TIMEOUT)
+                    except subprocess.TimeoutExpired:
+                        log_file.truncate(0)
+                        log_file.write(f"time out of {TIMEOUT} seconds reached\n")
+                        print(f"{file_to_run} - time out of {TIMEOUT} seconds reached")
+        finally:
+            # Clean up the temporary file created for this specific bit-width and original file
+            if os.path.exists(file_to_run):
+                os.remove(file_to_run)
+                print(f"Deleted temporary file: {file_to_run}")
+
+    elif benchmark == "instcombine":
```

Comment:
```suggestion
    elif benchmark == "instcombine":
```

We should have an `else` branch that throws an error, in case the user passes nothing.

Reply:
done